### PR TITLE
Handling POST in utils

### PR DESCRIPTION
**Pull request by angelika233**
## Summary
Added handling POST requests in `handle_api_response()` and added it to `Epicor` source

## Importance
`Epicor` can use function from `utils.py`  

## Checklist

This PR:

- [ ] adds new tests (if appropriate)
- [x] updates docstrings for any new functions or function arguments (if appropriate)
- [x] updates `CHANGELOG.md` with a summary of the changes

### CODE REVIEW

File: viadot/sources/epicor.py [hunk 1]
```diff
@@ -232,7 +234,7 @@ def generate_token(self) -> str:
             "password": self.credentials["password"],
         }
 
-        response = requests.request("POST", url, headers=headers)
+        response = handle_api_response(url=url, headers=headers, method="post")
```
Comment (lzuchowska) [inline viadot/sources/epicor.py:237]: Let's keep the convention for names of HTTP methods and capitalize them:
`get` -> `GET`
`post` -> `POST`

File: viadot/utils.py [hunk 2]
```diff
@@ -33,8 +35,11 @@ def handle_api_response(
         params (Dict[str, Any], optional): the request params also includes parameters such as the content type. Defaults to None.
         headers: (Dict[str, Any], optional): the request headers. Defaults to None.
         timeout (tuple, optional): the request times out. Defaults to (3.05, 60 * 30).
+        method (Literal ["get", "post"], optional): REST API medthod to use. Defaults to "get".
```
Comment (lzuchowska) [inline viadot/utils.py:38]: Typo:
`REST API medthod`

**Review by lzuchowska — CHANGES_REQUESTED** [review]
Please update the string and fix typo.

**Review by lzuchowska — APPROVED** [review]
Looks fine to me

File: viadot/utils.py [hunk 3]
```diff
@@ -54,13 +63,23 @@ def handle_api_response(
...
+            )
+        elif method == "POST":
+            response = session.post(
+                url,
+                auth=auth,
+                params=params,
+                headers=headers,
+                timeout=timeout,
+                data=body,
+            )
```
Comment (Trymzet) [inline viadot/utils.py:82]: You can use `session.request(method=method)`, then you don't need this big elif.

File: viadot/utils.py [hunk 4]
```diff
@@ -33,8 +35,11 @@ def handle_api_response(
         params (Dict[str, Any], optional): the request params also includes parameters such as the content type. Defaults to None.
         headers: (Dict[str, Any], optional): the request headers. Defaults to None.
         timeout (tuple, optional): the request times out. Defaults to (3.05, 60 * 30).
+        method (Literal ["GET", "POST"], optional): REST API method to use. Defaults to "GET".
+        body (str, optional): Data to send using post method. Defaults to None.
```
Comment (Trymzet) [inline viadot/utils.py:39]: I think HTTP methods are capitalized by convention, so it should be "using POST method".

File: viadot/utils.py [hunk 5]
```diff
@@ -33,8 +35,11 @@ def handle_api_response(
...
 
     Raises:
+        ValueError: raises when 'method' parameter value is unavailable
```
Comment (Trymzet) [inline viadot/utils.py:42]: I would say "hasn't been specified" rather than "is unavailable".

**Review by Trymzet — CHANGES_REQUESTED** [review]
Micro changes only

**Review by Trymzet — APPROVED** [review]
Looks good :)

**Pull request merged**